js/dom.js

```diff
@@ -0,0 +1,124 @@
+import { fetchData } from './api.js'
+import { getUserTrips, getUserExpenditures } from './index.js'
+
+const travelInfo = document.getElementById('travel-info')
+const userDataButton = document.getElementById('get-user-data-button')
+const travelerDetails = document.getElementById('travelerDetails')
+const tripDetails = document.getElementById('tripDetails')
+const expenditureDetails = document.getElementById('expenditureDetails')
+const destinationDetails = document.getElementById('destinationDetails');
+
+userDataButton.addEventListener('click', displayUserData)
+
+const loadData = () => {
+    return Promise.all([
+        fetchData('http://localhost:3001/api/v1/travelers/'),
+        fetchData('http://localhost:3001/api/v1/trips/'),
+        fetchData('http://localhost:3001/api/v1/destinations/')])
+}
+
+function displayUserData() {
+    const userID = parseInt(document.getElementById('userID').value)
+    if (!userID) {
+        travelInfo.innerText = 'Please enter a User ID'
+        return;
+    }
+    loadData()
+        .then(([travelers, trips, destinations]) => {
+            const travelerData = travelers.travelers.find(traveler => traveler.id === userID)
+            displayTravelerData(travelerData)
+
+            const tripData = getUserTrips(userID, trips, destinations)
+            displayTripData(tripData)
+
+            const expenditures = getUserExpenditures(userID, trips, destinations,)
+            displayExpenditureData(expenditures)
+
+            displayDestinationData(destinations.destinations)
+        })
+        .catch(error => console.error('There was a problem loading your data:', error))
+}
+
+const displayTravelerData = (data) => {
+    if (data) {
+        travelerDetails.innerHTML = `
+            <p>Name: ${data.name}</p>
+            <p>Your Travel Penchant: ${data.travelerType}</p>
+        `
+    } else {
+        travelerDetails.innerHTML = `${data.error}`
```

Comment:
Since you're passing in fetch data, I don't think this else will invoke as your catch from fetchData function will show instead